File: FinBalancer.Api/Models/InAppNotification.cs
namespace FinBalancer.Api.Models;

/// <summary>Obavijest prikazana korisniku u aplikaciji.</summary>
public class InAppNotification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Type { get; set; } = string.Empty; // AccountLinkInvite, BudgetAlert, etc.
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; }
    /// <summary>Opcionalni ID vezanog entiteta (npr. linkId za AccountLinkInvite).</summary>
    public string? RelatedId { get; set; }
    public string? ActionRoute { get; set; } // e.g. /linked-accounts
}

File: FinBalancer.App/lib/models/in_app_notification.dart
/// Obavijest u aplikaciji.
class InAppNotification {
  final String id;
  final String type;
  final String title;
  final String body;
  final bool isRead;
  final DateTime createdAt;
  final String? relatedId;
  final String? actionRoute;

  InAppNotification({
    required this.id,
    required this.type,
    required this.title,
    required this.body,
    required this.isRead,
    required this.createdAt,
    this.relatedId,
    this.actionRoute,
  });

  factory InAppNotification.fromJson(Map<String, dynamic> json) {
    return InAppNotification(
      id: json['id'] as String,
      type: json['type'] as String? ?? '',
      title: json['title'] as String? ?? '',
      body: json['body'] as String? ?? '',
      isRead: json['isRead'] as bool? ?? false,
      createdAt: DateTime.parse(json['createdAt'] as String),
      relatedId: json['relatedId'] as String?,
      actionRoute: json['actionRoute'] as String?,
    );
  }
}
